pkg/3scale/amp/operator/apicast_options_provider.go

```diff
@@ -60,6 +60,18 @@ func (a *ApicastOptionsProvider) setResourceRequirementsOptions() {
 		a.apicastOptions.ProductionResourceRequirements = v1.ResourceRequirements{}
 		a.apicastOptions.StagingResourceRequirements = v1.ResourceRequirements{}
 	}
+
+	// DeploymentConfig-level ResourceRequirements CR fields have priority over
+	// spec.resourceRequirementsEnabled, overwriting that setting when they are
+	// defined
+	if a.apimanager.Spec.Apicast.ProductionSpec.Resources != nil {
```

Comment:
`a.apimanager.Spec.Apicast.ProductionSpec` is not nil always?

Reply:
Yes.

We initialize it in the `setDefaults` method for the CR